File: brickport-infrastructure/src/services/in-memory/queries/player-queries.cs
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using BrickPort.Services.Queries;

namespace BrickPort.Infrastructure.Services.InMemory.Queries
{
    public class InMemoryPlayerQueries : IPlayerQueries
    {
        private readonly InMemoryDataStore _dataStore;

        public InMemoryPlayerQueries(InMemoryDataStore dataStore) => _dataStore = dataStore;

        public async Task<IEnumerable<Player>> GetAsync() => await Task.FromResult(
            _dataStore.Players.Select(x => new Player() 
            {
                PlayerId = x.Item1,
                PlayerName = x.Item2
            })
        );

        public async Task<Player> GetAsync(string id) => await Task.FromResult(
            (await GetAsync()).Single(summary => summary.PlayerId == id)
        );

    }
}

File: brickport-ui/src/brickport-services/queries/player-queries.ts
export interface IPlayer {
    playerId: string | null;
    playerName: string | null;
}

export class Player implements IPlayer {
    playerId!: string | null;
    playerName!: string | null;
    constructor(state?: IPlayer) {
        if (state) {
            Object.assign(this, state);
        }
        this.playerId = this.playerId || null;
        this.playerName = this.playerName || null;
    }
}

export interface IPlayerQueries {
    getAsync(): Promise<IPlayer[]>;
    getAsync(id: string): Promise<IPlayer>;
}